mongodump/mongodump_test.go

```diff
@@ -1977,3 +2004,83 @@ func TestFailDuringResharding(t *testing.T) {
 
 	})
 }
+
+func TestMongoDumpColumnstoreIndexes(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+	log.SetWriter(ioutil.Discard)
+
+	session, err := testutil.GetBareSession()
+	if err != nil {
+		t.Errorf("could not get session: %v", err)
+	}
+	fcv := testutil.GetFCV(session)
+	if cmp, err := testutil.CompareFCV(fcv, "6.3"); err != nil || cmp < 0 {
+		t.Skip("Requires server with FCV 6.3 or later")
+	}
+
+	require.NoError(t, setUpMongoDumpTestData())
+
+	// Create Columnstore indexes.
+	for _, colName := range testCollectionNames {
+		require.NoError(t, setUpColumnstoreIndex(testDB, colName))
+	}
+
+	md := simpleMongoDumpInstance()
+	md.ToolOptions.Namespace.DB = testDB
+	md.OutputOptions.Out = "dump"
+
+	require.NoError(t, md.Init())
+	require.NoError(t, md.Dump())
+
+	defer tearDownMongoDumpTestData()
+
+	path, err := os.Getwd()
+	require.NoError(t, err)
+
+	dumpDir := util.ToUniversalPath(filepath.Join(path, "dump"))
+	dumpDBDir := util.ToUniversalPath(filepath.Join(dumpDir, testDB))
+	require.True(t, fileDirExists(dumpDir))
+	require.True(t, fileDirExists(dumpDBDir))
+
+	defer os.RemoveAll(dumpDir)
+
+	c1, err := countNonIndexBSONFiles(dumpDBDir)
+	require.NoError(t, err)
+
+	c2, err := countMetaDataFiles(dumpDBDir)
+	require.NoError(t, err)
+	require.Equal(t, c1, c2)
+
+	metaFiles, err := getMatchingFiles(dumpDBDir, ".*\\.metadata\\.json")
+	require.NoError(t, err)
+	require.Greater(t, len(metaFiles), 0)
+
+	oneMetaFile, err := os.Open(util.ToUniversalPath(filepath.Join(dumpDBDir, metaFiles[0])))
```

Comment:
Would it be difficult to have it check the columnstore indexes for all relevant collections?

Reply:
No, just updated the test to check all collections.